packages/netlify-cms-core/src/reducers/__tests__/collections.spec.js

```diff
@@ -34,4 +53,20 @@ describe('collections', () => {
       }),
     );
   });
+
+  it('should inject unknown fields from an entry as hidden fields', () => {
+    expect(
+      selectFields(
+        collection,
+        '',
+        fromJS({
+          ...entry,
+          data: {
+            title: 'Test entry',
+            tag: 'Unkown field value',
+          },
+        }),
+      ),
+    ).toEqual(fromJS([{ name: 'title', widget: 'string' }, { name: 'tag', widget: 'hidden' }]));
+  });
```

Comment:
This would probably happen in a different test file, but we should test the output as well, to ensure the values persist in the output.

Reply:
Made two test cases, one for entries.persistEntry and another for editorialWorkflow.persistUnpublishedEntry to check for the existence and values of an unconfigured field.